chapter_natural-language-processing/machine-translation.md

```diff
@@ -29,12 +29,44 @@ def process_one_seq(seq_tokens, all_tokens, all_seqs, max_seq_len):
 
 # 使用所有的词来构造词典。并将所有序列中的词变换为词索引后构造NDArray实例
 def build_data(all_tokens, all_seqs):
-    vocab = text.vocab.Vocabulary(collections.Counter(all_tokens),
```

Comment:
vocab = text.vocab.Vocabulary(collections.Counter(all_tokens), reserved_tokens=[PAD, BOS, EOS])

should unchanged